pkg/lib/harness/llm-harness.go

```diff
@@ -181,8 +183,26 @@ func PrintLogs(configHome string, w io.Writer) error {
 		return fmt.Errorf("error reading log file: %w", err)
 	}
 	defer logFile.Close()
-	if _, err = io.Copy(w, logFile); err != nil {
-		return fmt.Errorf("failed to print log file: %w", err)
+	if !tail {
+		if _, err = io.Copy(w, logFile); err != nil {
+			return fmt.Errorf("failed to print log file: %w", err)
+		}
+	} else {
+		reader := bufio.NewReader(logFile)
+		for {
+			line, err := reader.ReadString('\n')
+			if err != nil {
+				if err == io.EOF {
+					time.Sleep(1 * time.Second)
+				} else {
+					return fmt.Errorf("failed to print log file: %w", err)
+				}
+			}
+			stringReader := strings.NewReader(line)
+			if _, err = io.Copy(w, stringReader); err != nil {
+				return fmt.Errorf("failed to print log file: %w", err)
```

Comment:
This is unncessary overhead you can write the line here instead
```suggestion
	        if _, err := w.Write([]byte(line)); err != nil {
                return fmt.Errorf("failed to write to output: %w", err) 
```